src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputFormattedNoWhitespaceBeforeAnnotations.java

```diff
@@ -0,0 +1,35 @@
+package com.google.checkstyle.test.chapter4formatting.rule462horizontalwhitespace;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+
+/** some javadoc. */
+public class InputFormattedNoWhitespaceBeforeAnnotations {
+
+  @Target(ElementType.TYPE_USE)
+  @interface NonNull {}
+
+  @NonNull int @NonNull [] @NonNull [] fiel1; // ok until #8205
```

Comment:
@Zopsss, please recheck what is wrong here and why formatter did not change it.

Reply:
if formatter  is ok, why we have issue in checkstyle to expect violation there

Reply:
From [4.6.2 Horizontal whitespace](https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace):

> a single ASCII space also appears in the following places only.
>
> 10. Between a type annotation and `[]` or `....`

We require whitespace between type-use annotations and `[]` or `...`, we cannot detect type-use annotation as described at: https://github.com/checkstyle/checkstyle/issues/15416

original code example:

https://github.com/checkstyle/checkstyle/blob/28c50973f23158bad3e8cc2cef1b36320af84952/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputNoWhitespaceBeforeAnnotations.java#L12-L15

https://github.com/checkstyle/checkstyle/blob/28c50973f23158bad3e8cc2cef1b36320af84952/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputNoWhitespaceBeforeAnnotations.java#L32-L34

There is already whitespace between `[]` , `...` and Type-use annotation `@NonNull`, so the code is already correct. Formatter dont need to format it. IMO the until comment is misleading.

Though we cannot detect code like:

```java
import java.lang.annotation.ElementType;
import java.lang.annotation.Target;

/** some javadoc. */
public class HspaceTest {

  @Target(ElementType.TYPE_USE)
  @interface NonNull {
  }

  @NonNull int @NonNull[] @NonNull[] fiel1; // false-negative

  public void foo3(final char @NonNull[] param) { // false-negative
  }

  void test2(String @NonNull... param) { // false-negative
  }
}
```

```
$ java -jar checkstyle-10.17.0-all.jar -c google_checks.xml HspaceTest.java 
Starting audit...
Audit done.
```